src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts

```diff
@@ -60,6 +60,20 @@ export const useWorkspaceForm = ({ application, defaultValues, onSubmit }: Works
     formIdRef.current = workspaceHtmlIdGenerator();
   }
 
+  const handleUseCasesChange = useCallback(
+    (newUseCases: string[]) => {
+      setFeatureConfigs((previousFeatureConfigs) => {
+        return [
+          ...previousFeatureConfigs.filter(
+            (featureConfig) => !isUseCaseFeatureConfig(featureConfig)
+          ),
+          ...newUseCases.map((useCaseItem) => getUseCaseFeatureConfig(useCaseItem)),
+        ];
```

Comment:
Can we simply do this?
```suggestion
        return appendDefaultFeatureIds(newUseCases.map((useCaseItem) => getUseCaseFeatureConfig(useCaseItem)));
```

Reply:
The append default feature ids logic is in the useState part above. It will be added when component did mount. Do you mean our feature configs will only includes use case or default feature ids? Current implementation is more for compatible with more static feature ids in the future.